k9mail/src/main/java/com/fsck/k9/helper/FileHelper.java

```diff
@@ -86,23 +87,16 @@ public static void touchFile(final File parentDir, final String name) {
 
     private static void copyFile(File from, File to) throws IOException {
         FileInputStream in = new FileInputStream(from);
+        FileOutputStream out = new FileOutputStream(to);
         try {
-            FileOutputStream out = new FileOutputStream(to);
-            try {
-                byte[] buffer = new byte[1024];
-                int count = -1;
-                while ((count = in.read(buffer)) > 0) {
-                    out.write(buffer, 0, count);
-                }
-            } finally {
-                out.close();
+            byte[] buffer = new byte[1024];
+            int count = -1;
+            while ((count = in.read(buffer)) > 0) {
+                out.write(buffer, 0, count);
             }
         } finally {
-            try { in.close(); } catch (Throwable ignore) {}
-        }
-
-        if (!from.delete()) {
-            Log.d(K9.LOG_TAG, "Unable to delete file: " + from.getAbsolutePath());
+            IOUtils.closeQuietly(out);
```

Comment:
You only want to use `closeQuietly()` on an `OutputStream` when an exception already occurred. Otherwise you mask exceptions that happen during a `close()`. Those should be treated the same as exceptions during `write()` calls.